File: Exercism Codes/C#/BeerSong.cs
using System;
using System.Text;

public static class BeerSong
{
    public static string Recite(int startBottles, int takeDown)
    {
        StringBuilder song = new StringBuilder();
        for (int i = startBottles, j = 0; j < takeDown; i--, j++)
        {
            song.Append((i == 0 ? "No more" : i) + (i == 1 ? " bottle" : " bottles") + " of beer on the wall, " + (i == 0 ? "no more" : i) + (i == 1 ? " bottle" : " bottles") + " of beer.\n" +
                    (i == 0 ? "Go to the store" : "Take " + (i == 1 ? "it" : "one") + " down") + " and " + (i == 0 ? "buy some more, 99" : "pass it around, " + (i == 1 ? "no more" : i - 1)) + (i == 2 ? " bottle" : " bottles") + " of beer on the wall.\n\n");
        }
        return song.ToString().Trim();
    }
}

File: Exercism Codes/Kotlin/BeerSong.kt
object BeerSong {
    fun verses(startBottles: Int, takeDown: Int): String {
        return (takeDown..startBottles).reversed().joinToString("\n") {
            "${
                if (it == 0) "No more" else it
            } bottle${
                if (it == 1) "" else "s"
            } of beer on the wall, ${
                if (it == 0) "no more" else it
            } bottle${
                if (it == 1) "" else "s"
            } of beer.\n${
                if (it == 0) "Go to the store" else "Take ${
                    if (it == 1) "it" else "one"
                } down"
            } and ${
                if (it == 0) "buy some more" else "pass it around"
            }, ${
                if (it - 1 == 0) "no more" else if (it == 0) 99 else it - 1
            } bottle${
                if (it - 1 == 1) "" else "s"
            } of beer on the wall.\n"
        }
    }
}
